### PR TITLE
Refactor DS inference API. No longer need replace_method.

### PR DESCRIPTION
**Pull request by awan-10**
This PR is aimed at simplifying our inference API both in terms of usage and the engine.py code.

We only support 3 cases:
1- User-specified injection policy dict. for tensor-parallelism
2- DeepSpeed kernel injection
3- New automatic tensor parallelism (that creates the injection policy dict. automatically)

@RezaYazdaniAminabadi -- please verify the above and review this PR. 

This PR will also address the user issue: https://github.com/microsoft/DeepSpeed/issues/2602

### CODE REVIEW

File: docs/_tutorials/inference-tutorial.md [hunk 1]
```diff
@@ -111,8 +110,7 @@ generator = pipeline('text-generation', model='EleutherAI/gpt-neo-2.7B',
 generator.model = deepspeed.init_inference(generator.model,
                                            mp_size=world_size,
                                            dtype=torch.float,
-                                           replace_method='auto',
-					   replace_with_kernel_inject=True)
+					            replace_with_kernel_inject=True)
```
Comment (molly-smith) [inline docs/_tutorials/inference-tutorial.md:113]: Extra indents here

**Comment by lekurile** [comment]
Just leaving this here as a reminder for us to remove `replace_method` from our `DeepSpeedExamples` repo as well, e.g.:
https://github.com/microsoft/DeepSpeedExamples/blob/b0afe97c93edd20551de3bf1ac0fd6cbb3e1cdcb/inference/huggingface/text-generation/inference-test.py#L20

**Comment by lekurile** [comment]
Just leaving this here as a reminder for us to remove `replace_method` from our `DeepSpeedExamples` repo as well, e.g.:
https://github.com/microsoft/DeepSpeedExamples/blob/b0afe97c93edd20551de3bf1ac0fd6cbb3e1cdcb/inference/huggingface/text-generation/inference-test.py#L20

**Review by lekurile — APPROVED** [review]

**Review by mrwyattii — APPROVED** [review]
LGTM

**Review by molly-smith — APPROVED** [review]

**Pull request merged**